crates/oxc_cli/src/lint/runner.rs

```diff
@@ -115,6 +130,18 @@ impl LintRunner {
         }
     }
 
+    fn dump_rules(&self) {
+        let mut stdout = BufWriter::new(std::io::stdout());
+        for (category, rules) in &self.registered_rules {
```

Comment:
We should sort the names somewhere, or is it already sorted?

Reply:
How should they be sorted? Could we just have a static const vec that holds our desired order and sort here before we iterate?

Reply:
We should be able to clone the names here and then sort them.

Reply:
You can also utilize `BTreeSet`

Reply:
It seems they're already in the order defined by the enum. Is that OK?